include/deal.II/lac/vector.h

```diff
@@ -258,6 +258,32 @@ class Vector : public Subscriptor
   virtual void reinit (const size_type N,
                        const bool      omit_zeroing_entries=false);
 
+  /**
+   * Same as above, but will preserve the values of vector upon resizing.
+   * If we new size is bigger, we will have
+   * \f[
+   * \mathbf V \rightarrow
+   * \left(
+   * \begin{array}{c}
+   * \mathbf V   \\
+   * \mathbf 0
+   * \end{array}
+   * \right)
+   * \f]
+   * whereas if the desired size is smaller, then
+   * \f[
+   * \mathbf V_1 \leftarrow
+   * \left(
+   * \begin{array}{c}
+   * \mathbf V_1   \\
+   * \mathbf V_2
+   * \end{array}
+   * \right)
```

Comment:
We already have the second case if the vector is shrinked...

Reply:
that's true, but it's only the case if we have `omit_zeroing=false`, whereas I can not _encode_ the first one with the same function `reinit(size, bool)`. So I decided for a new function instead and then used the same name in LAPACKFullMatrix

Reply:
Sure, we do have that, I was more thinking about adding some more description pointing out this analogy in case someone wonders.